### PR TITLE
Update zap.NewJSON to use functional options

### PR DESCRIPTION
**Pull request by prashantv**
Default options are:
- Info
- Write logs to stdout
- Write logging errors to stderr
- No fields

Log levels implement the option interface so you can pass them directly.

Added examples for NewJSON that show default vs passing options.

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+type Option interface {
+	apply(*jsonLogger) error
+}
+
+// optionFunc wraps a func so it satisfies the Option interface.
+type optionFunc func(*jsonLogger) error
+
+func (f optionFunc) apply(jl *jsonLogger) error {
+	return f(jl)
+}
```
Comment (akshayjshah) [inline options.go]: :+1:

File: options.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+// This allows any Level to be used as an option.
+func (l Level) apply(jl *jsonLogger) error {
+	*jl.level = int32(l)
+	return nil
+}
```
Comment (akshayjshah) [inline options.go]: Level is atomic (though we clearly can't trigger a race from the current options application site). Just for consistency, can we use `jl.SetLevel` here instead?

Comment (prashantv) [inline options.go]: Good idea, cleans up the code. Done

**Comment by akshayjshah** [comment]
lgtm

**Pull request merged**